### PR TITLE
add task7

### PR DESCRIPTION
**Pull request by anton-latyshonok**
without logs

### CODE REVIEW

File: task7/task7.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    getPage(skip = DEFAULT_SKIP, top = DEFAULT_TOP, filterConfig = {}) {
+        if (skip < 0 || skip > this._posts.length) {
+            return [];
+        }
+
+        if (top < 0) {
+            return [];
+        }
+
+        const slicedPosts = this._posts.sort((firstPost, secondPost) => secondPost.createdAt - firstPost.createdAt).slice(skip, top + skip);
```
Comment (ghost) [inline task7/task7.js:199]: Incorrect order: you  need to filter posts first and then sort the result.

File: task7/task7.js [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+    constructor(posts) {
+        this._posts = posts;
+    }
+
+    static validate(post) {
+        if (!post) {
+            return false;
+        }
+
+        const { description, id, createdAt, author, photoLink='' } = post;
```
Comment (ghost) [inline task7/task7.js:170]: '+

File: task7/task7.js [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+        }
+
+        const slicedPosts = this._posts.sort((firstPost, secondPost) => secondPost.createdAt - firstPost.createdAt).slice(skip, top + skip);
+
+        if (typeof filterConfig !== 'object') {
+            return slicedPosts;
+        }
+
+        return slicedPosts.filter((post) => {
+            return (!filterConfig.id || filterConfig.id === post.id)
```
Comment (ghost) [inline task7/task7.js:206]: hashtags?

File: task7/task7.js [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            return false;
+        }
+
+        const currentPost = this.getPost(id);
+
+        if (!currentPost) {
+            return false;
+        }
+
+        const newPost = { ...currentPost, post };
```
Comment (ghost) [inline task7/task7.js:234]: '+

File: task7/task7.js [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+            if (!this.add(newPost)) {
+                invalidedPosts = [...invalidedPosts, newPost];
+
+                return invalidedPosts;
+            }
+        }, [])
+    }
+
+
+    getPage() {
```
Comment (ghost) [inline task7/task7.js:266]: Are you sure this is works :) 
I think no, keys (methods) are duplicated))

Comment (anton-latyshonok) [reply task7/task7.js:266]: yea
i duplicate only logs (they will be almost the same for everyone 👍 )

**Pull request merged**